tremor-cli/src/test/tag.rs

```diff
@@ -125,31 +124,42 @@ impl TagFilter {
     }
 
     pub(crate) fn join(&self, tags: Option<Vec<String>>) -> TagFilter {
-        let mut includes: Vec<String> = self.includes.iter().cloned().collect();
-        let excludes: Vec<String> = self.excludes.iter().cloned().collect();
+        let mut includes: Tags = self.includes.iter().cloned().collect();
+        let excludes: Tags = self.excludes.iter().cloned().collect();
         if let Some(mut tags) = tags {
             includes.append(&mut tags);
         }
         TagFilter::new(excludes, includes)
     }
 }
 
+// The intention is to find all tag files in the directories between base and other
+// basically if base is /a/b/c and other is /a/b/c/d/e/f we want to look for:
+// - /a/b/c/tags.json
+// - /a/b/c/d/tags.json
+// - /a/b/c/d/e/tags.json
+// - /a/b/c/d/e/f/tags.json
 pub(crate) fn resolve(base: &Path, other: &Path) -> Result<TagFilter> {
-    if let Ok(rel) = util::relative_path(base, other) {
-        let mut base = base.to_string_lossy().to_string();
-        let tags_file = format!("{}/tags.json", &base);
+    if other.strip_prefix(base).is_ok() {
+        let tags_file = base.join("tags.json");
         let mut tags = TagFilter::new(vec![], vec![]);
         tags = tags.join(Some(maybe_slurp_tags(&tags_file)));
-        for dirname in rel.split('/') {
-            base = format!("{}/{}", base, dirname);
-            let tags_file = format!("{}/tags.json", &base);
+        let depth = other.components().count() - base.components().count();
+        let walker = globwalk::glob_builder("tags.json")
```

Comment:
I don't thinkt that is going to work the way we want, could we add a test to make sure? I think this would find all tags.json relative to the base name of the executable.

It also would probably find something like /a/b/c/x/tag.json

Reply:
Forgot to add the base :facepalm: